modules/qr_code.py

```diff
@@ -9,11 +9,12 @@
 
 
 class QRCode:
-    def __init__(self, base_url, qr_cache_path, max_size) -> None:
+    def __init__(self, base_url, qr_cache_path, max_size, qr_image_path) -> None:
```

Comment:
lets make the parameter optional
```suggestion
    def __init__(self, base_url, qr_cache_path, max_size, qr_image_path = None) -> None:
```